packages/hdwallet-native/src/binance.ts

```diff
@@ -79,20 +79,22 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
 
         const client = new BncClient("https://dex.binance.org"); // broadcast not used but available
         await client.chooseNetwork("mainnet");
-        await client.setPrivateKey(privateKey, Number.isInteger(Number(msg.account_number)));
+        const haveAccountNumber = Number.isInteger(Number(msg.account_number));
+        if (haveAccountNumber) await client.setAccountNumber(Number(msg.account_number));
+        await client.setPrivateKey(privateKey, haveAccountNumber);
         await client.initChain();
 
-        const addressFrom = msg.tx?.msgs?.[0]?.inputs?.[0]?.address;
-        const addressFromVerify = client.getClientKeyAddress()
+        const addressFrom = msg.tx.msgs[0].inputs[0].address;
```

Comment:
The type specification on `msg` requires that this path exist, and if it doesn't a hard-fail is preferable to feeding `undefined` to the Binance SDK.